projects/igniteui-angular/src/lib/query-builder/query-builder-tree.component.html

```diff
@@ -6,11 +6,65 @@
     <igx-icon family="default" name="close"></igx-icon>
 </ng-template>
 
+<ng-template #selectFromTemplate>
+    <div class="igx-filter-tree__inputs" [style.display]="(isInEditMode() && this.entities.length > 0 && this.entities[0].name) ? 'flex' : 'none'">
+        <igx-select #entitySelect
+            type="box"
+            (selectionChanging)="onEntitySelectChanging($event)"
+            [overlaySettings]="entitySelectOverlaySettings"
+            [ngModel]="selectedEntity"
+            [style.display]="isInEditMode() ? 'block' : 'none'"
+        >
+            <label igxLabel>{{ this.resourceStrings.igx_query_builder_select_entity }}</label>
+            <igx-select-item *ngFor="let entity of entities" [value]="entity">
+                {{entity.name}}
+            </igx-select-item>
+        </igx-select>
+
+        <igx-combo
+            #selectedReturnFieldsCombo
+            [itemsMaxHeight]="250"
+            [data]="fields"
+            [displayKey]="'field'"
+            [valueKey]="'field'"
+            [disabled]="!selectedEntity"
+            [(ngModel)]="selectedReturnFields"
+            [overlaySettings]="returnFieldSelectOverlaySettings"
+            searchPlaceholder="{{ this.resourceStrings.igx_query_builder_search }}"
+            [style.display]="isInEditMode() ? 'block' : 'none'"
+            (selectionChanging)="onReturnFieldSelectChanging($event)"
+        >
+            <ng-template igxComboHeader>
+                <div
+                    class="igx-drop-down__item igx-drop-down__item--query-builder"
+                    (click)="onSelectAllClicked($event)"                
+                    [ngClass]="{'igx-drop-down__item--selected': this.selectedEntity && this.selectedReturnFields && this.selectedEntity.fields.length === this.selectedReturnFields.length}"
+                >
+                    <igx-checkbox
+                            [checked]="this.selectedEntity && this.selectedReturnFields && this.selectedEntity.fields.length === this.selectedReturnFields.length"
```

Comment:
To be safe, we should probably check the fields before getting the length (as we do for indeterminate):
```suggestion
                            [checked]="this.selectedEntity && this.selectedReturnFields && this.selectedEntity.fields?.length === this.selectedReturnFields.length"
```

Reply:
Since this line is not part of this PR, maybe it's better to make the change in the most upper branch query-builder-improvements and trickle down from there. I've posted the suggestion there